### PR TITLE
Feature/gift entry  form save

### PR DESCRIPTION
**Pull request by vmartinov-sforg**
# Critical Changes

# Changes
 * add support for Saving the form rendered

# Issues Closed

# New Metadata

# Deleted Metadata

# Testing Instructions
* create new template with the Template builder. Name the template 'Single Gift Entry Template'. Use the url part /lightning/cmp/c__geTemplateBuilderPlaceholder appended to your instance url.
* Open the Form Render with the url part /lightning/cmp/c__geFormRendererPlaceholder appended to your instance url.
* Check if the fields in the Template are loaded.
* Enter data and Save
* Check if all the populated fields saved  to the opportunity and related objects.

# Not Included
* Lookup fields are not supported at this time

### CODE REVIEW

**Comment by vmartinov-sforg** [comment]
**lurch: attach W-037287

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HoEDQA0" target="_blank">W-037287</a>

File: src/lwc/geFormService/geFormService.js [hunk 1]
```diff
@@ -74,9 +78,53 @@ class GeFormService {
...
+                .then((result) => {
+                    resolve(result);
+                })
+                .catch(error => {
+                    console.error(JSON.stringify(error));
+                });
+        });
+    }
+
+    handleSave(sectionList) {
```
Comment (bdvorachek) [inline src/lwc/geFormService/geFormService.js]: Could use a docstring here

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 2]
```diff
@@ -28,6 +31,40 @@ export default class GeFormRenderer extends LightningElement {
...
+        });
+    }
+
+    // change showSpinner to the opposite of its current value
+    toggleSpinner() {
+        this.showSpinner = !this.showSpinner;
+    }
 
+    // Expose the labels to use in the template
+    label = { messageLoading };
```
Comment (bdvorachek) [inline src/lwc/geFormRenderer/geFormRenderer.js]: I think this should be further up, towards the other variable declarations.

File: src/lwc/geFormRenderer/geFormRenderer.html [hunk 3]
```diff
@@ -6,4 +6,14 @@
...
     </lightning-layout>
-</template>
\ No newline at end of file
+
+    <div if:true={showSpinner} class="spinnerContainer">
+        <lightning-spinner alternative-text={label.messageLoading} variant="brand" size="large"></lightning-spinner>
+    </div>
+
+    <div class="slds-docked-form-footer" style="height: 50px;">
+        <lightning-button label="Cancel" title="Cancel" onclick={handleCancel} class="slds-m-left_x-small"></lightning-button>
```
Comment (bdvorachek) [inline src/lwc/geFormRenderer/geFormRenderer.html]: Lets use labels for save/cancel strings

**Comment by vmartinov-sforg** [comment]
@bdvorachek  I made the requested changes. Pls let me know if I there's anything else I missed.

**Pull request merged**